File: 0169-majority-element/0169-majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        
        int ans = nums[0];
        int cnt = 0;

        for(int i : nums) {
            if(ans == i)
                cnt++;
            else
                cnt--;
            
            if(cnt == 0) {
                ans = i;
                cnt = 1;
            }
        }

        return ans;
    }
};

File: 0169-majority-element/0169-majority-element.ts
function majorityElement(nums: number[]): number {

    let ans = 0;
    let cnt = 0;

    for(let i = 0; i < nums.length; i++) {
        if(cnt == 0) {
            ans = nums[i];
            cnt++;
        } else if(ans == nums[i])
            cnt++;
        else
            cnt--;
    }

    return ans;
};